python/tvm/topi/cuda/scatter_elements.py

```diff
@@ -20,6 +20,188 @@
 from tvm import te, tir
 from ..utils import ceil_div, get_const_int
 from ..math import cast
+from .nms import atomic_add
+
+
+def gen_scatter_add_1d_atomic(data, indices, updates, out, axis, _):
```

Comment:
Ah this is something I think I might have missed. Within indices along a particular axis, do we expect all indices to be unique? Otherwise I believe something similar must be done with all operators.

Reply:
Yes, potentially we can. The problem is that quick search did not give me any atomic operation like atomic_add, moreover it is not supported by some architectures ("vulkan", "metal"). I postponed this task due to it requires more time for investigation and implementation but does not have any priority.